src/wasm.h

```diff
@@ -1514,6 +1514,7 @@ class RefI31 : public SpecificExpression<Expression::RefI31Id> {
   RefI31() = default;
   RefI31(MixedArena& allocator) {}
 
+  Shareability share;
```

Comment:
Can this not be part of the `type`? We don't seem to have such a field on anything else.

Reply:
Hmm, I guess it could.

Reply:
See the last commmit for this change.